src/components/DayPickerRangeController.jsx

```diff
@@ -642,7 +642,8 @@ export default class DayPickerRangeController extends React.Component {
 
   isHovered(day) {
     const { hoverDate } = this.state || {};
-    return isSameDay(day, hoverDate);
+    const { focusedInput } = this.props;
+    return focusedInput && isSameDay(day, hoverDate);
```

Comment:
I think with the new perf rearchitecture in v11.0.0, we will also have to wrap https://github.com/chris-verclytte/react-dates/blob/c5e7fae7986518dec6cb4e2438d5f0a402d15605/src/components/DayPickerRangeController.jsx#L400-L425 in a `if (focusedInput)` conditional in this PR.